### PR TITLE
Don't scroll outside of focusable popup

### PR DESCRIPTION
**Pull request by igordmn**
Fixes https://github.com/JetBrains/compose-jb/issues/1346

### CODE REVIEW

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt [hunk 1]
```diff
@@ -360,7 +360,7 @@ class ComposeScene internal constructor(
 
     private fun SkiaBasedOwner?.isAbove(
         targetOwner: SkiaBasedOwner?
-    ) = list.indexOf(this) > list.indexOf(targetOwner)
+    ) = this != null && targetOwner != null && list.indexOf(this) > list.indexOf(targetOwner)
```
Comment (olonho) [inline compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt:363]: We iterate list twice.
Why not just 
```
var seenThis = false
list.forEach {
  if it == this  seenThis = true
  if it == targetOwner return@forEach  !seenThis 
}
```

**Review by olonho — APPROVED** [review]

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt (hunk 1 above)
Comment (igordmn) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt:363]: > We iterate list twice.

It doesn't affect performance, and looks simplier.

And iterating through list and doing 5 operations usually takes the same amount of time as iterating through list 5 times, doing the same operation in each iteration (iteration itself is rare bottleneck).

Comment (olonho) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt:363]: Not in this case, I guess, original code  computed both indices unconditionally, and in solution above we always stop when encounter targetOwner. Probably not too important for small lists, anyway.

**Review by olonho — APPROVED** [review]

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt (hunk 1 above)
Comment (igordmn) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt:363]: > Probably not too important for small lists, anyway.

Yes, this is my though as well, better to make code simpler :)

**Pull request merged**